Controllers/Timesheets.php

```diff
@@ -66,17 +62,36 @@ public function get(array $params): void
             if ($date = $this->timesheetsExporter->getDateTime($criteria['dateTo'] ?? null)) {
                 $filename .= '_' . $date->format(\DateTimeInterface::ATOM);
             }
-            if ('excel' === $format) {
-                $filename .= '.excel';
-            }
-            $filename .= '.csv';
+            $filename .= '.' . $format;
             $this->timesheetsExporter->export(
-                $params,
+                $query,
                 options: [
-                'filename' => $filename,
-                'format' => $format,
+                    'filename' => $filename,
+                    'format' => $format,
                 ]
             );
         }
+
+        throw new HttpResponseException(Frontcontroller::redirect(BASE_URL . '/errors/error404'));
+    }
+
+    /**
+     * Get context from query string parameters.
+     *
+     * @return array
+     *   id: id
+     *   format: format
+     *   query: the rest of the parameters
+     */
+    private function getContext(array $params): array
+    {
+        $leantime = array_filter(
+            $params,
+            static fn (mixed $key) => in_array($key, ['id', 'act', 'request_parts'], true),
+            ARRAY_FILTER_USE_KEY
+        );
+        $query  = array_diff_key($params, $leantime);
+
+        return [$leantime, $query];
```

Comment:
Should this just return `$leantime['id]` and `$query`?

`id` is the only thing used from the `$leantime` array and format later fetched from `$query`.

Reply:
Good point. However, the idea behind this function is to make a general way to separate Leantime parameters from controller parameters (query string parameters), and to make it reusable by future controllers. Leantime may already have a built-in way to do this.